### ISSUE TITLE
Asset fingerprinting regression

### ISSUE THREAD

**Issue by ilucin** [ember-cli/ember-cli#7149]
After upgrading to 2.13.3 I've noticed super weird regression bug in asset fingerprinting.

Somewhere in my code I have something like:
`
{url: 'assets/c3.js'}
{url: 'assets/dropzone.js'}
`

When running `ember build --environment=production` i get this:
`
{url:"/assets/c3-5326c4dedefdea21697981ac7029f23c.js}
{url:"/assets/dropzone-b17e693e9af725242f0ed6b156038b79.js}
`

But after upgrading to 2.13.3 I got this:
`
{url:"/assets/c3-5326c4dedefdea21697981ac7029f23c.js}
{url:"/assets/dropzone-ed4fcb056fe9ff48b2fbac86069506c3-5326c4dedefdea21697981ac7029f23c.js"}
`

Notice the wrong concatenation on dropzone file.

After a small investigation I've discovered that #7077 did this with config:
`minifiyJS: {options: {compress: {output: {semicolons: false}}}}`

I've no idea why but it's probably something that should be checked out.

cc @rwjblue 

---

Output from `ember version --verbose && npm --version && yarn --version`:
```
ember-cli: 2.13.3
http_parser: 2.7.0
node: 8.0.0
v8: 5.8.283.41
uv: 1.11.0
zlib: 1.2.11
ares: 1.10.1-DEV
modules: 57
openssl: 1.0.2k
icu: 58.2
unicode: 9.0
cldr: 30.0.3
tz: 2016j
os: darwin x64
5.0.1
0.24.6
```

**Comment by rwjblue** [comment]
@ilucin - thanks for tracking this down! I am somewhat surprised, but its very helpful that you tracked it back to https://github.com/ember-cli/ember-cli/pull/7077.  

Can you provide a demo repo that I could use to track this down?

**Comment by ilucin** [comment]
@rwjblue here's the repo -> https://github.com/ilucin/reproduce-ember-cli-fingerprint-bug

I've tried to reduce the context as much as I could. There's an in-repo-addon that creates lib bundles that are loaded lazily in the (real) app. One weird thing I also discovered is that if I put "dropzone.js" instead of "dropzone.min.js" into public tree of "lib-bundles" the bug doesn't occur. This means  broccoli-uglify-sourcemap somehow affects broccoli-asset-rev I guess? 

And also, what I mentioned, removing `minifiyJS: {options: {compress: {output: {semicolons: false}}}}` default option from ember-cli also fixes it.

I was just using build command to reproduce the wrong fingerprint:
`ember build --environment=production; cat dist/assets/reproduce-ember-cli-fingerprint-bug-*.js | grep assets.test.com`

 I really don't have much debugging experience with ember-cli so I can't help much. Maybe I'm doing something weird/stupid with this "lib-bundles" in-repo-addon?
